.ci/macos-steps.yaml

```diff
@@ -15,8 +15,8 @@ steps:
     sudo xcode-select --switch /Applications/Xcode_10.1.app/Contents/Developer
     unset BOOST_ROOT
     pip install cython numpy pandas zipp
-    brew update
     brew install openblas armadillo boost
+    brew update
```

Comment:
Hm, does that make any difference, because this will have no effect on the packages that are installed before, armadillo, etc. But easy to test, can you remove that one?

Reply:
Hi @zoq if we install armadillo before updating brew we are going to install :
```
==> Installing dependencies for armadillo: eigen, hwloc, open-mpi, arpack, szip and hdf5
==> Installing armadillo dependency: eigen
==> Downloading https://homebrew.bintray.com/bottles/eigen-3.3.7.mojave.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/68/683c2dd898245f61c298d1f2675885e7c67ec7e18f6665df1ec56f088bd670f4?__gda__=exp=1583974502~hmac=144e8f5107c18055b3ef857a5540be92b81ca053cd0c1614887dd8c29131f863&response-content-disposition=attachment%3Bfilename%3D%22eigen-3.3.7.mojave.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX19uHX-Abm1Z-SO_l_jTcN6W1I9z4doq3YUsqzHNJ73vx26C1sTdrnEm4MTYjdo5An6QfR0UYVWyY5YijoK3NU33RqDBwPxXAJd1OoVW7jZMOoeHA7avGDJAsC3k-6BwYlBEFDngtxZu9g&response-X-Checksum-Sha1=8546974d10de9a2dde2cffc5237bc2bddc8e3c68&response-X-Checksum-Sha2=683c2dd898245f61c298d1f2675885e7c67ec7e18f6665df1ec56f088bd670f4
==> Pouring eigen-3.3.7.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/eigen/3.3.7: 487 files, 6.5MB
==> Installing armadillo dependency: hwloc
==> Downloading https://homebrew.bintray.com/bottles/hwloc-2.1.0.mojave.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/df/df6180858171e5345d517cb5d7bace1f0f33fd63a84180ec591f2530465d7172?__gda__=exp=1583974506~hmac=23a05d6a7bc57a4acde65960a727df7263d6b8f0167c9ee0a40a631c47b10228&response-content-disposition=attachment%3Bfilename%3D%22hwloc-2.1.0.mojave.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX19GrVPRP-66j97FeWuCPXERO15_EiAumhsEqiMSRHpFhb95aTBPSx80QOLVR6JlX2phM6n26T2Ki_q5r_BsGX34Zh_bhKmmD7lgPSv_zoaMEH_76E9Jafm7diiewbyo7r2H4nSKjr-RfQ&response-X-Checksum-Sha1=9fc98a890fe5f142fb8f908d104ae8a6bea32190&response-X-Checksum-Sha2=df6180858171e5345d517cb5d7bace1f0f33fd63a84180ec591f2530465d7172
==> Pouring hwloc-2.1.0.mojave.bottle.tar.gz
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
==> Summary
🍺  /usr/local/Cellar/hwloc/2.1.0: 881 files, 9.5MB
==> Installing armadillo dependency: open-mpi
==> Downloading from https://akamai.bintray.com/16/1600986b4774f6081191fe616cc70690d0174c4a32f4d7f280285ddc39970437?__gda__=exp=1583974509~hmac=aa09b8d33d9ead89d43405ae8f6b195607b2dba331c2f2002505913caf612847&response-content-disposition=attachment%3Bfilename%3D%22open-mpi-4.0.3.mojave.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX1_E13TBoOBkKrzH0fi5c9dJN5OjRDyIepayz86DqBGzjwy9TR9pp8iyudl97jNAQUpCeq9bBSs1UKnOWtcxSb6drxjpa533Y313OG_IXbvT2cPHpVjGZwXCt0Pp0VYkkg_N9WUBJIpFhA&response-X-Checksum-Sha1=05c7c0cfe04d3a33262375f3686664b774d26b0b&response-X-Checksum-Sha2=1600986b4774f6081191fe616cc70690d0174c4a32f4d7f280285ddc39970437
==> Pouring open-mpi-4.0.3.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/open-mpi/4.0.3: 752 files, 10.7MB
==> Installing armadillo dependency: arpack
==> Downloading https://homebrew.bintray.com/bottles/arpack-3.7.0_4.mojave.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/db/db006f0681a1d280da53ac6ac60904a51e0d8158708d20fdcc4e6404289db850?__gda__=exp=1583974514~hmac=d7353e7a7be9db45f322dcf696672721f7efd8887934a32b26c646f6e7e7e827&response-content-disposition=attachment%3Bfilename%3D%22arpack-3.7.0_4.mojave.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX19fGIN0Mxd9mgCdehEsdSiPirnhJFg5MBzmBJvPdj9NwOD5-ribL0aQLESErobZ0ifs5lvWdX42zLNrBDD2a3ZuayCzhQ7Sz1Hq4qMrCB_jbduZp-0EaKAKx4wjN8imcNVCcdKF6YsxcA&response-X-Checksum-Sha1=255b9c0c16928938edc5d9c2a4eb3e087b59ac17&response-X-Checksum-Sha2=db006f0681a1d280da53ac6ac60904a51e0d8158708d20fdcc4e6404289db850
==> Pouring arpack-3.7.0_4.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/arpack/3.7.0_4: 37 files, 3.2MB
==> Installing armadillo dependency: szip
==> Downloading https://homebrew.bintray.com/bottles/szip-2.1.1_1.mojave.bottle.tar.gz
==> Pouring szip-2.1.1_1.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/szip/2.1.1_1: 11 files, 109.4KB
==> Installing armadillo dependency: hdf5
==> Downloading https://homebrew.bintray.com/bottles/hdf5-1.10.6.mojave.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/ae/ae200d2122b45c5a08c3f1b8d2d141eac3fbe362c3f330f81003146e81c0fef5?__gda__=exp=1583974518~hmac=e9cb7cc0bfcfb2478fa907b926d0ca0fb163f5949df0d89d35520059fea76f87&response-content-disposition=attachment%3Bfilename%3D%22hdf5-1.10.6.mojave.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX1-RogNUvtuzEsQB85PgssCWDGpFQNIAQPYPvhxEssqsfc4DzkCkP-_TX9KzeZA8GMjbct3DULHB7tJqkTAoREoppwdwvhlbr2XHPrAVRW_TpJz6adcQRgfeeWSK1_zMJQlPPvsFXRsHrw&response-X-Checksum-Sha1=71c239e44dd33fad668ea5e4d55fddaaca4bcf09&response-X-Checksum-Sha2=ae200d2122b45c5a08c3f1b8d2d141eac3fbe362c3f330f81003146e81c0fef5
==> Pouring hdf5-1.10.6.mojave.bottle.tar.gz
Warning: hdf5 dependency gcc was built with a different C++ standard
library (libstdc++ from clang). This may cause problems at runtime.
🍺  /usr/local/Cellar/hdf5/1.10.6: 258 files, 14.7MB
==> Installing armadillo
==> Downloading https://homebrew.bintray.com/bottles/armadillo-9.850.1.mojave.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/0a/0a0a345d802ce2fd8c6e5914116d738839952ce3e6dd978bf961d8ff9d9d269b?__gda__=exp=1583974521~hmac=289ee817a78f5a6296c7fbb70df1f558c02c4dd105c8f1d8836cb820793412d4&response-content-disposition=attachment%3Bfilename%3D%22armadillo-9.850.1.mojave.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX19ZFSFXNoA3fArJ_EFD_6715uJKuYACh-Gv8EmIqUqx3wRAy-cqcY3N5DoIfFuSMMa9bjT9TAv7ccsdFashMOpPhILrXzimutW79HQ7MSCZy4b6pzdFPoOGqp-AmKm6C7MMGf4sl38FmA&response-X-Checksum-Sha1=ad5423c15042e3b1093b4abe19fdc5b9137a1ae5&response-X-Checksum-Sha2=0a0a345d802ce2fd8c6e5914116d738839952ce3e6dd978bf961d8ff9d9d269b
```
Else if we are updating brew before installing armadillo then we are to going install:
```
==> Installing armadillo dependency: open-mpi
==> Downloading https://homebrew.bintray.com/bottles/open-mpi-4.0.3.high_sierra.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/71/71df1f5047b812b68f8b28f7a6c713f42389691b2269408c8d839f1922ab9e5c?__gda__=exp=1584054775~hmac=19480699b46a56c90ed773fc16a84d908d383b69e984285e8969bf410ed34f8d&response-content-disposition=attachment%3Bfilename%3D%22open-mpi-4.0.3.high_sierra.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX1_vyCHDcPzKPY5aPS8PQh-IsmonrmN0kVCDgf8beNyCG1uvFTJ4Aqm7xU60fumYu1EF9y58koBKMdweOn0djDuSXZLhk8fePIwWOZUmZiJqwOwEhuzZtbOl8l7qtZha3I7MD1em5ADk8Q&response-X-Checksum-Sha1=03b9ebf6272807ecb55671bdb3634c7ac0e011a2&response-X-Checksum-Sha2=71df1f5047b812b68f8b28f7a6c713f42389691b2269408c8d839f1922ab9e5c
==> Pouring open-mpi-4.0.3.high_sierra.bottle.tar.gz
🍺  /usr/local/Cellar/open-mpi/4.0.3: 752 files, 10.8MB
==> Installing armadillo dependency: arpack
==> Downloading https://homebrew.bintray.com/bottles/arpack-3.7.0_4.high_sierra.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/51/510b490c32d7adf344c655133c83b74d17fcdba51ccafe6e0a5a1a7f9899f3b1?__gda__=exp=1584054782~hmac=5c5851fe2d2702259f6c07456f4fe159f7f27daf4dc6e74a9cb9b602338da52f&response-content-disposition=attachment%3Bfilename%3D%22arpack-3.7.0_4.high_sierra.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX1_DZqlTbzeCS2uaBKxfvSYG3D4e0R0wwGHO26mLxn7vYSfrcEjfzZ2E_Ijh0rd_LvjKB-CZnWMxouqzSKkaFfUkKx78rhamM6CyW5ooRorRaM0jveH1h7dbvlPASWuVD3RoFCWG9gGK8g&response-X-Checksum-Sha1=5577be3e595c6daec822ce182b9278a52c9182f1&response-X-Checksum-Sha2=510b490c32d7adf344c655133c83b74d17fcdba51ccafe6e0a5a1a7f9899f3b1
==> Pouring arpack-3.7.0_4.high_sierra.bottle.tar.gz
🍺  /usr/local/Cellar/arpack/3.7.0_4: 37 files, 3.2MB
==> Installing armadillo dependency: szip
==> Downloading https://homebrew.bintray.com/bottles/szip-2.1.1_1.high_sierra.bottle.tar.gz
==> Pouring szip-2.1.1_1.high_sierra.bottle.tar.gz
🍺  /usr/local/Cellar/szip/2.1.1_1: 11 files, 108.7KB
==> Installing armadillo dependency: hdf5
==> Downloading https://homebrew.bintray.com/bottles/hdf5-1.12.0.high_sierra.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/88/88a4048d5d26ccea95574726fd874d8b69409301ed6a0f36d9c695168b3fc144?__gda__=exp=1584054788~hmac=898662c0957911d7789f0d632de0dff022ea97c1a3b1a01fbf7bb8388f5b5aa6&response-content-disposition=attachment%3Bfilename%3D%22hdf5-1.12.0.high_sierra.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX18tlielKgj0NNqzWAuHB474iPtvk7l1o6xA2jHcL8GqxMXqX8ADG3GQffeo-wzGqRVILOO1ZvwLXPFBa_GNhMHHZWK4QGPi-5Ukdc3nOkLoKvR-NYp4WzDqebLDSew3qSyhXzXM-y8_DQ&response-X-Checksum-Sha1=12c5a08a10c753526e369cba40aca8746c5aa277&response-X-Checksum-Sha2=88a4048d5d26ccea95574726fd874d8b69409301ed6a0f36d9c695168b3fc144
==> Pouring hdf5-1.12.0.high_sierra.bottle.tar.gz
Warning: hdf5 dependency gcc was built with a different C++ standard
library (libstdc++ from clang). This may cause problems at runtime.
🍺  /usr/local/Cellar/hdf5/1.12.0: 268 files, 16.4MB
==> Installing armadillo
==> Downloading https://homebrew.bintray.com/bottles/armadillo-9.850.1_1.high_sierra.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/4b/4b3c8bd9e1d57a331dedb35dcbb401c16b700e97dc94c4c2a459acf108561577?__gda__=exp=1584054792~hmac=0107f8e943ee29a3250f9b108c736ad05649a2963028fb44de57517593201294&response-content-disposition=attachment%3Bfilename%3D%22armadillo-9.850.1_1.high_sierra.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX19St-_JdjY07sk3XSeMhI-ZoPIkWmMgvUyZr1OevhCl4Mr2d5-YV59NvZN_TF17pOWEssrh35vHs98cE50rc7_cWVC8ztxk8wDyKc5zZwrlzDuTHu2NA0vptJj9Eegyp1scSAfuiM9om2592kJDCS5tKb1ETpEp-jo&response-X-Checksum-Sha1=71ddfa802e2210a138272fc633b8a2e70812cf04&response-X-Checksum-Sha2=4b3c8bd9e1d57a331dedb35dcbb401c16b700e97dc94c4c2a459acf108561577
==> Pouring armadillo-9.850.1_1.high_sierra.bottle.tar.gz
```
Here we are installing hdf5-1.12.0 which i think so resulting into build failure.
And I think so brew update is also important because it is solving some julia issue.
Thank you.

Reply:
Right, but in this case `brew install openblas armadillo boost` comes before `brew update` so it will have no effect, it might have an effect on `brew cask install julia` but I don't think it's needed.

Reply:
Hi @zoq If I remove `brew update`  I got this error :

```
==> Downloading https://julialang-s3.julialang.org/bin/mac/x64/1.3/julia-1.3.1-mac64.dmg
Error: Checksum for Cask 'julia' does not match.
Expected: b3df0bfde44c16688c140ac94358fcae8c3e4dcb14a68576054e667370cf86f1
  Actual: 6bc0f815a6b533ac9069ed6018c2c4d86208d5ec57c67c254f2f70e8f500107f
    File: /Users/runner/Library/Caches/Homebrew/downloads/0eeae5f5ffba37cb842a27f2745af3a648d2d0b5611ac837f73da84b800b5478--julia-1.3.1-mac64.dmg
To retry an incomplete download, remove the file above.
If the issue persists, visit:
  https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/checksum_does_not_match_error.md
==> Verifying SHA-256 checksum for Cask 'julia'.
==> Note: Running `brew update` may fix SHA-256 checksum errors.
```
Should I still remove `brew update` and take a try.?

Reply:
Let's see what happens if we remove it in the ci, job, if it fails we can reinclude it, does that sounds good? Also, can you rebase against the current master branch, to enable the ci job?

Reply:
It worked you are great @zoq :)

Reply:
Nah, thanks for testing it out.